apollo-ios-codegen/Sources/IR/IR+SelectionSet.swift

```diff
@@ -160,3 +162,17 @@ public class SelectionSet: Hashable, CustomDebugStringConvertible {
   }
 
 }
+
+extension LinkedList where T == ScopeCondition {
+  var containsDeferredFragment: Bool {
+    var node: Node? = last
+    var deferCondition = node?.value.deferCondition
+
+    while node?.previous != nil && deferCondition == nil {
+      node = node?.previous
+      deferCondition = node?.value.deferCondition
+    }
+
+    return deferCondition != nil
```

Comment:
This is a small improvement in my opinion:

```suggestion
    do {
      guard node?.value.deferCondition == nil else {
        return true
      }

      node = node?.previous      
    } while node != nil

    return false
```